### ISSUE TITLE
Can we pass in a pre-trained word embeddings (i.e. Flair/FastText) in the TransformerDocumentEmbedding?

### ISSUE THREAD

**Issue by junhua** [flairNLP/flair#1590]
RT. Thanks!

**Comment by alanakbik** [comment]
Hello @junhua that's not possible as the transformers are pre-trained and have their own subword embedding layers. You could combine `TransformerWordEmbeddings` and other word embeddings and put for instance `DocumentRNNEmbeddings` on top, but I never evaluated this so not sure if this works better or worse than just using `TransformerDocumentEmbeddings`.

**Comment by junhua** [comment]
I tried using TransformerDocumentEmbedding for a document classification tasks and it performed significantly poorer than stacked embedding (FastText+Flair_news_fw + Flair_news_bw) with RNN document embedding (LSTM with 2 layers 30 hidden states). It's like ~60% vs ~90% for F1. 

Anyway, isn't it possible to pool a stacked embedding into the same input format for transformer document embedding and fine-tune the pre-trained transformer LM? Not sure if I'm making sense...

**Comment by alanakbik** [comment]
Unfortunately that is not possible because the transformer has its own embedding layer that it was pre-trained on, so that part cannot be exchanged afterwards. 

When you train with TransformerDocumentEmbeddings, you have to use Adam, a very low learning rate and only a few epochs. So this is different from training LSTMEmbeddings. Here is an example: 

```python
            trainer = ModelTrainer(model, corpus, optimizer=torch.optim.Adam)
            trainer.train(
                f'path/to/output/folder',
                learning_rate=3e-5, 
                max_epochs=5,
                min_learning_rate=3e-6,
            )
```

**Comment by junhua** [comment]
Yeap I used one of the recommended LR (5e-5) and Adam to train for 5 epochs. The performance is not satisfactory. Here's how it looks like:

<img width="880" alt="image" src="https://user-images.githubusercontent.com/3516360/81327743-51e75800-90ce-11ea-93b9-2c29d1ca63e5.png">

Here's how BiLSTM looks like:

<img width="810" alt="image" src="https://user-images.githubusercontent.com/3516360/81327957-ab4f8700-90ce-11ea-9b2d-2ead8f602851.png">

<img width="912" alt="image" src="https://user-images.githubusercontent.com/3516360/81327928-9f63c500-90ce-11ea-88bd-1783a689ca4e.png">

I use HuggingFace's transformer package to train too, and it performs very well with the same hyperparam, the result is like this (precision, recall, f1, acc):

<img width="778" alt="image" src="https://user-images.githubusercontent.com/3516360/81328325-49435180-90cf-11ea-937b-51be6c25ae0c.png">

**Comment by alanakbik** [comment]
In your training code, your missing the value `min_learning_rate=3e-6,`. Could you add this and run it again?

**Comment by junhua** [comment]
Turn out well! What difference does it make since Flair will take the best model within 5 epoch?

**Comment by alanakbik** [comment]
It's basically a bug, I'll fix this so you don't always have to specify the min learning rate. Do you get similar scores now as HuggingFace?

**Comment by junhua** [comment]
Yes it is. Wonderful :)

**Comment by alanakbik** [comment]
Great :) Yes we use AutoModel internally in the class.
